etc/docker/entrypoint.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
```

Comment:
The purpose of this entrypoint file is somewhat opaque to me. Can I get some additional context? The majority of the file seems to be about dealing with the cases where `testbot-uid` or `testbot-gid` are provided as arguments, but I don't see anything in this repo that does that.

Reply:
@Otterverse would explain better than me but my understanding is that it is a workaround permission issues. Docker can make virtual permissions, which mean that if a user has a UID different from testbot then we may not be able to created files or created files may not be readable by said user. This is mainly relevant when using the image for compiling a local repository (for ex `docker run -it --rm --name esp32 --mount type=bind,source="$(pwd)"/docker,target=/src -v "$SSH_AUTH_SOCK:$SSH_AUTH_SOCK" -e SSH_AUTH_SOCK="$SSH_AUTH_SOCK" test-rust-image:latest --testbot-uid 1000 --testbot-gid 1000  '. "$IDF_PATH"/export.sh && . "$IDF_PATH"/../export-esp.sh && make build'`)

Reply:
This was copied from rdk basically, and the canon image that uses (which is in the SysOps repo.) It allows the "make canon-shell" (and other canon targets) to remap UID/GIDs internally. But it's not used directly by the CI.

As mentioned to @npmenard in conversation, I'm in the process of creating a canon utility that will handle this at runtime, so this could be removed entirely if it's not needed for the next short while until that utility is ready for release. (Without the makefile targets to make use of this, is't a complex command to run, so unlikely to be used.)